pkg/scheduler/plugins/predicates/cache.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2020 The Volcano Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package predicates
+
+import (
+	"fmt"
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog"
+
+	batch "volcano.sh/volcano/pkg/apis/batch/v1alpha1"
+)
+
+type predicateCache struct {
+	sync.Mutex
+	cache map[string]map[string]bool //key_1: nodename key_2:pod uid
+}
+
+// predicateCacheNew return cache map
+func predicateCacheNew() *predicateCache {
+	return &predicateCache{
+		cache: make(map[string]map[string]bool),
+	}
+}
+
+// getPodTemplateUID return pod template key
+func getPodTemplateUID(pod *v1.Pod) string {
+	uid, found := pod.Annotations[batch.PodTemplateKey]
+	if !found {
+		return ""
+	}
+
+	return uid
+}
+
+// PredicateWithCache: check the predicate result existed in cache
+func (pc *predicateCache) PredicateWithCache(nodeName string, pod *v1.Pod) (bool, error) {
+	podTemplateUID := getPodTemplateUID(pod)
+	if podTemplateUID == "" {
+		return false, fmt.Errorf("no anonation of volcano.sh/template-uid in pod %s", pod.Name)
+	}
+
+	pc.Lock()
+	defer pc.Unlock()
+	if nodeCache, exist := pc.cache[nodeName]; exist {
+		if result, exist := nodeCache[podTemplateUID]; exist {
+			klog.V(4).Infof("Predicate node %s and pod %s result %v", nodeName, pod.Name, result)
+			return result, nil
+		}
+	}
+
+	return false, fmt.Errorf("No information of node %s and pod %s in predicate cache", nodeName, pod.Name)
+}
+
+// UpdateCache update cache data
+func (pc *predicateCache) UpdateCache(nodeName string, pod *v1.Pod, fit bool) {
```

Comment:
suggest to return a bool flag to mark the update result

Reply:
here is a action that just filling the data map table, no logic of failure